apps/hash-api/src/graph/ensure-system-graph-is-initialized/system-webs-and-entities.ts

```diff
@@ -0,0 +1,350 @@
+        preferredName = "Linear Integration";
+      } else {
+        throw new Error(
+          `Unhandled web shortname ${webShortname} requires a display name for the machine actor specified`,
+        );
+      }
+
+      if (error instanceof NotFoundError) {
+        await createMachineActorEntity(context, {
+          machineAccountId: machineActorAccountId,
+          identifier: webShortname,
+          ownedById: accountGroupId as OwnedById,
+          displayName: preferredName,
+          shouldBeAbleToCreateMoreMachineEntities: false,
+          systemAccountId,
+        });
+
+        logger.info(
+          `Created machine actor entity for '${webShortname}' machine types, using accountId ${machineActorAccountId} in accountGroupId '${accountGroupId}`,
+        );
+      } else {
+        throw error;
+      }
+    }
+  }
+
+  /**
+   * Create the HASH Instance entity, which stores configuration settings for the instance
+   */
+  const authentication = { actorId: systemAccountId };
+  try {
+    await getHashInstance(context, authentication, {});
+  } catch (error) {
+    if (error instanceof NotFoundError) {
+      await createHashInstance(context, authentication, {});
+      logger.info("Created hashInstance entity");
+    } else {
+      throw error;
+    }
+  }
+
+  /**
+   * Create the HASH AI Machine actor and entity, which is added as needed to webs to run AI-related workflows.
+   */
+  try {
+    await getMachineActorId(context, authentication, {
+      identifier: "hash-ai",
+    });
+  } catch (error) {
+    if (error instanceof NotFoundError) {
+      const aiAssistantAccountId = await createAccount(
+        context,
+        authentication,
+        {},
+      );
+
+      const hashAccountGroupId = owningWebs.hash.accountGroupId;
+      if (!hashAccountGroupId) {
+        throw new Error(
+          `Somehow reached the point of creating the HASH AI machine actor without a hash accountGroupId`,
+        );
+      }
+
+      await createMachineActorEntity(context, {
+        identifier: "hash-ai",
+        machineAccountId: aiAssistantAccountId,
+        ownedById: hashAccountGroupId as OwnedById,
+        displayName: "HASH AI",
+        shouldBeAbleToCreateMoreMachineEntities: false,
+        systemAccountId,
+      });
+
+      logger.info("Created HASH AI entity");
+    } else {
+      throw error;
+    }
+  }
+
+  /**
+   * Mop up step: create web machine actors for existing webs – bots with permissions to add other bots to each existing web,
+   * and to create notifications that aren't tied to specific integrations (e.g. related to comments and @mentions).
+   *
+   * This step is only required to transition existing instances in Dec 2023, and can be deleted once they have been migrated.
+   */
+  const users = await getEntitiesByType(context, authentication, {
+    entityTypeId: "https://hash.ai/@hash/types/entity-type/user/v/1", // @todo this may need to change depending on migration strategy
+  });
```

Comment:
Yes - I started off with it in the migration file, but on a fresh instance we don't have the `hash` machine entity available until this point, because it can't be created until after the last migration file has run. In theory we could create it in the migration file but it started to feel like a lot of moving things around that would be long-term less satisfactory to avoid this short term hack.

The `hash` machine entity has to exist because we use it as part of giving other machines permissions, and since all that code is currently in `backend-utils` I didn't want to import the `systemAccountId` from `@apps/hash-api` to retrieve the actor, but instead by searching for the entity. The `createMachineActor` code could probably live separately in the API, but it again felt like a compromise which would would be worse in the long term.